st2actions/st2actions/runners/pythonrunner.py

```diff
@@ -69,13 +70,25 @@ def __init__(self, config=None):
         :type config: ``dict``
         """
         self.config = config or {}
-        # logger and datastore are assigned in PythonActionWrapper._get_action_instance
-        self.logger = None
+        self.logger = get_logger_for_action(action_name=self.__class__.__name__)
+
+        # action_service is late assigned post class instatiation in "setup()"
         self.datastore = None
 
+        # Flag which indicates if the class has been initialized and "setup()" has been called
+        self._initialized = False
+
+    def setup(self, action_service):
+        """
+        Method which performs additional class initialization.
+        """
+        self.action_service = action_service
+        self._initialized = True
+
     @abc.abstractmethod
     def run(self, **kwargs):
-        pass
+        if not self._initialized:
```

Comment:
Yeah, mentioned this on Slack - it's mostly pointless since most actions don't call parent run so I can remove it.